transport/src/main/java/io/netty/channel/DefaultChannelId.java

```diff
@@ -247,25 +247,36 @@ public int hashCode() {
     }
 
     @Override
-    public int compareTo(ChannelId o) {
-        return 0;
+    public int compareTo(DefaultChannelId o) {
+        if (this == o) {
+            // short circuit
+            return 0;
+        }
+        // lexicographic comparison
+        int len1 = data.length;
+        int len2 = o.data.length;
+        int len = Math.min(len1, len2);
+
+        int k = 0;
+        while (k < len) {
+            byte x = data[k];
+            byte y = o.data[k];
+            if (x != y) {
+                return x - y;
```

Comment:
consider adding a comment that overflow won't occur because the arithmetic is promoted to `int`.